testutils/src/main/java/com/microsoft/identity/internal/testutils/labutils/KeyVaultAuthHelper.java

```diff
@@ -0,0 +1,73 @@
+package com.microsoft.identity.internal.testutils.labutils;
+
+import com.microsoft.identity.common.internal.providers.keys.CertificateCredential;
+import com.microsoft.identity.common.internal.providers.keys.ClientCertificateMetadata;
+import com.microsoft.identity.common.internal.providers.keys.KeyStoreConfiguration;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftClientAssertion;
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsTokenRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenRequest;
+import com.microsoft.identity.internal.test.keyvault.Configuration;
+
+import java.io.IOException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+class KeyVaultAuthHelper extends ConfidentialClientHelper {
+
+    private final static String CLIENT_ID = "4bc6e96f-bd23-408f-8ecb-a7a7145463f9";
+    private final static String SCOPE = "https://vault.azure.net/.default";
+    private final static String CERTIFICATE_ALIAS = "AutomationRunner";
+    private final static String KEYSTORE_TYPE = "Windows-MY";
+    private final static String KEYSTORE_PROVIDER = "SunMSCAPI";
+    private final static String MSSTS_CLIENT_ASSERTION_AUDIENCE = "https://login.microsoftonline.com/microsoft.com/oauth2/v2.0/token";
+
+    private String mAccessToken;
+
+    private static KeyVaultAuthHelper sKeyVaultAuthHelper;
+
+    private KeyVaultAuthHelper() {
+        mAccessToken = null;
+    }
+
+    public static ConfidentialClientHelper getInstance() {
+        if (sKeyVaultAuthHelper == null) {
+            sKeyVaultAuthHelper = new KeyVaultAuthHelper();
+        }
+        return sKeyVaultAuthHelper;
+    }
+
+    @Override
+    public String getAccessToken() throws UnrecoverableKeyException, NoSuchAlgorithmException, KeyStoreException, NoSuchProviderException, IOException, InterruptedException, CertificateException {
```

Comment:
should we move this function and setupApiClientWithAccessToken to base class?

Reply:
we need to have an implementation for both the LAB API and the Key Vault so I had to separate it out.

Reply:
if we move `getAccessToken` to the base class, then we will need two member variables for access token in that base class. One to hold access token for key vault and other to hold access token for lab

Reply:
Addressed